src/retriever/data_tiers/tier_0/dgraph/result_models.py

```diff
@@ -13,25 +13,30 @@
 NODE_KEY_PATTERN = re.compile(r"(?:q\d+_)?node_(\w+)")
 
 
+def _strip_prefix(d: Mapping[str, Any], prefix: str | None) -> Mapping[str, Any]:
+    if not prefix:
+        return d
+    return { (k[len(prefix):] if k.startswith(prefix) else k): v for k, v in d.items() }
+
 # -----------------
 # Dataclasses
 # -----------------
 
 
 @dataclass(frozen=True, slots=True, kw_only=True)
-class PublicationsInfo:
-    """Information regarding publications."""
-
-    pmid: str
-    publication_date: str | None = None
-    sentence: str | None = None
-    subject_score: str | None = None
-    object_score: str | None = None
+class Source:
```

Comment:
The source model is missing some optional fields, see the [TRAPI RetrievalSource spec](https://github.com/NCATSTranslator/ReasonerAPI/blob/bc396712607d46a7be730abfaf885cf4a23fa25f/TranslatorReasonerAPI.yaml#L1700-L1756)

Reply:
Added the optional fields
- upstream_resource_ids
- source_record_urls